rts/lib/lua/src/lbaselib.cpp

```diff
@@ -236,10 +236,15 @@ static int luaB_next (lua_State *L) {
 
 
 static int luaB_pairs (lua_State *L) {
-  luaL_checktype(L, 1, LUA_TTABLE);
-  lua_pushvalue(L, lua_upvalueindex(1));  /* return generator, */
-  lua_pushvalue(L, 1);  /* state, */
-  lua_pushnil(L);  /* and initial value */
+  luaL_checkany(L, 1);
+  if (luaL_getmetafield(L, 1, "__pairs") == LUA_TNIL) {  /* no metamethod? */
+    lua_pushvalue(L, lua_upvalueindex(1));  /* return generator, */
```

Comment:
Perhaps this could have `luaL_checktype(L, 1, LUA_TTABLE);`? I'd expect that to produce a better error message ("error in pairs...") than letting it fail inside `next` (since the next is there implicitly).

Reply:
I think it failing inside `next` will be fine- it having failed in `pairs` will just be one level up the callstack

I can't move the `luaL_checktype(L, 1, LUA_TTABLE);` from `next` since that is sometimes used on it's own in Lua code, and checking the type twice in a row seems... silly (and also inefficient)